101-hub-and-spoke-sandbox/azuredeploy.json

```diff
@@ -0,0 +1,916 @@
+
+            ]
+         },
+         "dependsOn":[
+            "[resourceId('Microsoft.Network/virtualNetworks', variables('hubVnetName'))]"
+         ]
+      },
+      {
+         "comments":"Shared Subnet",
+         "type":"Microsoft.Network/virtualNetworks/subnets",
+         "name":"[concat(variables('hubVnetName'), '/', variables('sharedSubnetName'))]",
+         "apiVersion":"2018-02-01",
+         "scale":null,
+         "properties":{
+            "addressPrefix":"[variables('sharedSubnetPrefix')]",
+            "serviceEndpoints":[
+
+            ]
+         },
+         "dependsOn":[
+            "[resourceId('Microsoft.Network/virtualNetworks', variables('hubVnetName'))]"
+         ]
+      },
+      {
+         "comments":"DMZ Subnet",
+         "type":"Microsoft.Network/virtualNetworks/subnets",
+         "name":"[concat(variables('hubVnetName'), '/', variables('dmzSubnetName'))]",
+         "apiVersion":"2018-02-01",
+         "scale":null,
+         "properties":{
+            "addressPrefix":"[variables('dmzSubnetPrefix')]",
+            "serviceEndpoints":[
+
+            ]
+         },
+         "dependsOn":[
+            "[resourceId('Microsoft.Network/virtualNetworks', variables('hubVnetName'))]"
+         ]
+      },
+      {
+         "condition":"[equals(parameters('deployVpnGateway'), 'Yes')]",
+         "comments":"Gateway Subnet",
+         "type":"Microsoft.Network/virtualNetworks/subnets",
+         "name":"[concat(variables('hubVnetName'), '/', variables('gatewaySubnetName'))]",
+         "apiVersion":"2018-02-01",
+         "scale":null,
+         "properties":{
+            "addressPrefix":"[variables('gatewaySubnetPrefix')]",
+            "serviceEndpoints":[
+
+            ]
+         },
+         "dependsOn":[
+            "[resourceId('Microsoft.Network/virtualNetworks', variables('hubVnetName'))]"
+         ]
+      },
+      {
+         "comments":"Development Spoke Virtual Network (VNet)",
+         "type":"Microsoft.Network/virtualNetworks",
+         "name":"[variables('devSpokeVnetName')]",
+         "apiVersion":"2018-02-01",
+         "location":"[resourceGroup().location]",
+         "scale":null,
+         "properties":{
+            "addressSpace":{
+               "addressPrefixes":[
+                  "[variables('devSpokeVnetPrefix')]"
+               ]
+            },
+            "dhcpOptions":{
+               "dnsServers":[
+
+               ]
+            },
+            "subnets":[
+
+            ],
+            "virtualNetworkPeerings":[
+
+            ],
+            "enableDdosProtection":false,
+            "enableVmProtection":false
+         },
+         "dependsOn":[
+
+         ]
+      },
+      {
+         "comments":"Development Workload Subnet",
+         "type":"Microsoft.Network/virtualNetworks/subnets",
+         "name":"[concat(variables('devSpokeVnetName'), '/', variables('spokeWorkloadSubnetName'))]",
+         "apiVersion":"2018-02-01",
+         "scale":null,
```

Comment:
remove all of these "null" properties (won't comment on the rest but need to remove them)